components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/AbstractAuthorizationGrantHandler.java

```diff
@@ -852,17 +853,14 @@ private void clearExistingTokenFromCache(OAuthTokenReqMessageContext tokenMsgCtx
                                              AccessTokenDO existingTokenBean) {
 
         if (cacheEnabled) {
-            OAuth2AccessTokenReqDTO tokenReq = tokenMsgCtx.getOauth2AccessTokenReqDTO();
-
-            String scope = OAuth2Util.buildScopeString(tokenMsgCtx.getScope());
-            String consumerKey = tokenMsgCtx.getOauth2AccessTokenReqDTO().getClientId();
-            String authorizedUser = tokenMsgCtx.getAuthorizedUser().toString();
-            String authenticatedIDP = tokenMsgCtx.getAuthorizedUser().getFederatedIdPName();
             String tokenBindingReference = getTokenBindingReference(tokenMsgCtx);
 
-            OAuthCacheKey cacheKey =
-                    getOAuthCacheKey(scope, consumerKey, authorizedUser, authenticatedIDP, tokenBindingReference);
-            removeFromCache(cacheKey, tokenReq.getClientId(), existingTokenBean);
+            OAuthUtil.clearOAuthCache(existingTokenBean.getConsumerKey(), existingTokenBean.getAuthzUser(),
+                    OAuth2Util.buildScopeString(existingTokenBean.getScope()), tokenBindingReference);
+            OAuthUtil.clearOAuthCache(existingTokenBean.getConsumerKey(), existingTokenBean.getAuthzUser(),
+                    OAuth2Util.buildScopeString(existingTokenBean.getScope()));
+            OAuthUtil.clearOAuthCache(existingTokenBean.getConsumerKey(), existingTokenBean.getAuthzUser());
+            OAuthUtil.clearOAuthCache(existingTokenBean.getAccessToken());
```

Comment:
For a JWT access token, what do we get when we call existingTokenBean.getAccessToken(), is it the JWT or the hash of the JWT?

Reply:
When its from cache, it gives the JWT. When its from db, it gives the JTI.